frontend/src/App.vue

```diff
@@ -104,48 +107,57 @@ export default {
     }
   },
   watch: {
-    "currentPrompt.answers": {
+    "copyCurrentPromptAnswers": {
       deep: true,
       immediate: true,
-      handler() {
+      handler(newAnswers, oldAnswers) {
+        let isEqual = true
         // TODO: consider using debounce (especially for questions of type 'input') to limit roundtrips
-        const curentPromptAnswers = _.get(this.currentPrompt, "answers")
-        const questions = _.get(this.currentPrompt, "questions", []);
+        const questions = _.get(this, "currentPrompt.questions", []);
         _.forEach(questions, question => {
           const questionName = question.name
-          if (question._default === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "default"]).then(response => {
-              question.default = response
-              if (question.answer === undefined) {
-                question.answer = question.default
-              }
-            })
-          }
-          if (question.when === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "when"]).then(response => {
-              question.isWhen = response
-            })
-          }
-          if (question._message === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "message"]).then(response => {
-              question.message = response
-            })
+          
+          if (isEqual) {
+            const newAnswer = _.get(newAnswers, [questionName])
+            const oldAnswer = _.get(oldAnswers, [questionName])
+            isEqual = _.isEqual(newAnswer, oldAnswer)
           }
-          if (question._choices === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[curentPromptAnswers], questionName, "choices"]).then(response => {
-              question.choices = response
-            })
-          }
-          if (question.filter === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer], questionName, "filter"]).then(response => {
-              question.answer = response
-            })
-          }
-          if (question.validate === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer, curentPromptAnswers], questionName, "validate"]).then(response => {
-              question.isValid = (_.isString(response) ? false : response)
-              question.validationMessage = (_.isString(response) ? response : undefined)
-            })
+          
+          if (!isEqual) {
```

Comment:
conside check (question.isWhen !==false)

Reply:
In the inquirer js documentation they do not tell that when "when ===false" all other question parameters do not change. 
We might create a bug applying such an assumption.

Reply:
In the yo cli if when === false then question value is undefined.